instrumentation/net/http/labeler.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/label"
+)
+
+// Labeler is used to allow instrumented HTTP handlers to add custom labels to
+// the metrics recorded by the net/http instrumentation.
+type Labeler struct {
+	labels []label.KeyValue
+}
+
+// Add labels to a Labeler.
+func (l *Labeler) Add(ls ...label.KeyValue) {
+	l.labels = append(l.labels, ls...)
```

Comment:
Does this need to be guarded by a mutex? Having this be used by handlers that will run in their own goroutine it would seem so.